File: Frontend/ApiUniversidad/UniversidadAngular/src/app/models/alumnos-por-curso.ts
import { Alumno } from "./alumno";
import { Curso } from "./curso";

export class AlumnosPorCurso {

    id?: string;
    idCurso: string = '';
    idAlumno: string = '';
    fechaAlta: Date = new Date();
    idAlumnoNavigation: Alumno = new Alumno();
    idCursoNavigation: Curso = new Curso();
    
}

File: Backend/ApiUniversidad/ApiUniversidad/Models/AlumnosPorCurso.cs
using System;
using System.Collections.Generic;

namespace ApiUniversidad.Models;

public partial class AlumnosPorCurso
{
    public Guid Id { get; set; }

    public Guid IdCurso { get; set; }

    public Guid IdAlumno { get; set; }

    public DateTime FechaAlta { get; set; }

    public virtual Alumno IdAlumnoNavigation { get; set; } = null!;

    public virtual Curso IdCursoNavigation { get; set; } = null!;
}
